symfony/plugins/orangehrmTimePlugin/Service/TimesheetService.php

```diff
@@ -456,12 +465,28 @@ public function checkForOverlappingTimesheets($startDate, $endDate, $employeeId)
     }
 
     /**
-     * Add, Update Timesheet
      * @param Timesheet $timesheet
-     * @return boolean
+     * @param DateTime $date
+     * @return Timesheet
      */
-    public function saveTimesheet(Timesheet $timesheet) {
-
+    public function saveTimesheet(Timesheet $timesheet, DateTime $date): Timesheet
+    {
+        $accessFlowStateMachineService = new AccessFlowStateMachineService();
```

Comment:
this service initialization you can move to getter method in this class

Reply:
done